shared/src/main/java/uk/gov/di/authentication/shared/services/SessionService.java

```diff
@@ -113,17 +113,11 @@ public void deleteSessionFromRedis(String sessionId) {
 
     public Optional<Session> readSessionFromRedis(String sessionId) {
         try {
-            if (redisConnectionService.keyExists(sessionId)) {
-                return Optional.of(
-                        segmentedFunctionCall(
-                                "Deserialise session",
-                                () ->
-                                        OBJECT_MAPPER.readValue(
-                                                redisConnectionService.getValue(sessionId),
-                                                Session.class)));
-            } else {
-                return Optional.empty();
-            }
+            String serializedSession = redisConnectionService.getValue(sessionId);
```

Comment:
@ethanmills are we ok with removing the segmentedFunctionCall here?
Both `redisConnectionServic.getValue()` and `OBJECT_MAPPER.readValue()` call it anyway so we should still have good visibility?

Reply:
Yeah that's fine